src/SFML/Audio/Sound.cpp

```diff
@@ -117,9 +110,9 @@ void Sound::setPlayingOffset(Time timeOffset)
 
 
 ////////////////////////////////////////////////////////////
-const SoundBuffer* Sound::getBuffer() const
+const SoundBuffer& Sound::getBuffer() const
 {
-    return m_buffer;
+    return *m_buffer;
```

Comment:
If ever sound is required to have a buffer, why do we use a pointer still to reference the buffer?

Reply:
We can't use a reference because in `sf::Sound::setBuffer` we reassign the buffer. We could consider using a std::reference_wrapper. It has the semantics we want. It can't be null yet does support reassignment. It's just a fancy wrapper around a raw pointer.

Reply:
https://github.com/ChrisThrasher/SFML/commit/fc5f0c1083561d431aee2a0a6dfcb19b57610cef

Reply:
Having a reference instead would make the Sound type non-copyable because references cannot be reassigned.

Reply:
I mean, if we can guarantee that the pointer can never be invalidated or `nullptr` then we can keep it. The issue I see with the change is, that we remove any null-checks, yet the keep the pointer. So if we have an invalid pointer, you could get some ugly crashes.